### ISSUE TITLE
x/tools/go/loader: needs to be updated to stdlib go/types

### ISSUE THREAD

**Issue by sbinet** [golang/go#11861]
hi,

with the upcoming go-1.5 and the import of (a modified version of) the x/tools/go/types package into the stdlib, `golang.org/x/tools/go/loader` needs to be updated.

the following program will fail to compile:
```go
package main

import (
	"fmt"
	"go/build"
	"go/token"
	"go/types"
	"log"
	"os"

	"golang.org/x/tools/go/loader"
)

func main() {
	path := os.Args[1]
	fmt.Printf(">>> inspecting [%s]...\n", path)

	cwd, err := os.Getwd()
	if err != nil {
		log.Panicf("error: %v\n", err)
	}

	pkg, err := build.Import(path, cwd, 0)
	if err != nil {
		log.Panicf("error: %v\n", err)
	}

	conf := loader.Config{
		Fset: token.NewFileSet(),
	}

	prog, err := conf.Load()
	if err != nil {
		log.Panicf("error: %v\n", err)
	}

	var p *types.Package = prog.Created[0].Pkg
	fmt.Printf("pkg=%v\n", p)
}
```

with the following error:
```sh
sh> go run foo.go github.com/go-python/gopy/_examples/hi
# command-line-arguments
./foo.go:39: cannot use prog.Created[0].Pkg (type *"golang.org/x/tools/go/types".Package) as type *"go/types".Package in assignment
```

**Comment by sbinet** [comment]
FYI, I have been testing this against golang/tools@1c460ff and golang/go@d23973d.

**Comment by josharian** [comment]
This is one of the unfortunate things about having two versions of the same packages. If this were to get fixed, then folks using the x/tools go/types would have breakage the other direction.

/cc @alandonovan @griesemer

**Comment by ianlancetaylor** [comment]
We should be able to use build tags in x/tools to make this work.

**Comment by sbinet** [comment]
using a go-1.5 build tag sounds good.
I can send a CL if nobody beats me to it.

**Comment by alandonovan** [comment]
I agree this needs to happen, but we cannot change the tools repo until after Go 1.5 has been released and has settled for at least a while.  My plan is to update the whole tools repo in early September so that it no longer uses tools/go/types (though of course that package will remain there).

**Comment by dsymonds** [comment]
If build tags are used, there's no need to wait for Go 1.5. This is also not tied to the release.

**Comment by adonovan** [comment]
Fully one third of the ~350 *.go files in the tools repo import golang.org/x/tools/go/types.  Are you suggesting that I make go1.5-tagged duplicates of all those files, and maintain them in parallel during several months of anticipated heavy churn?  Or something else?

**Comment by dsymonds** [comment]
I didn't realise it was that many files. Yeah, build tags seem inappropriate. Let's just do the change after the release.

**Comment by shurcooL** [comment]
> My plan is to update the whole tools repo in early September so that it no longer uses tools/go/types (though of course that package will remain there).

@adonovan, has that been done already, or not yet? If not, is there an updated estimate for when it'll happen?

**Comment by alandonovan** [comment]
I (finally) sent https://go-review.googlesource.com/#/c/18207/ for review today.  I'll submit it after the holiday since it will require that all clients using tip (not go1.5) update to `go/types` too.

**Comment by shurcooL** [comment]
This issue has been resolved via https://github.com/golang/tools/commit/542ffc7e75cf45ac7ad456664a710cd96338fda6 and can be closed.
